lib/bom.gradle

```diff
@@ -21,7 +21,27 @@ configure(projectsWithFlags('bom')) {
                 }
                 return "${a.ext.artifactId}".compareTo("${b.ext.artifactId}")
             }.each { p ->
-                api "${p.group}:${p.ext.artifactId}:${p.version}"
+                if (rootProject.ext.has('bomGroups')) {
+                    def bomGroups = rootProject.ext.bomGroups
+                    if (!(bomGroups instanceof Map)) {
+                        throw new IllegalStateException("bomGroups must be a Map: ${bomGroups}")
+                    }
+
+                    if (!bomGroups.containsKey(project.path)) {
+                        api "${p.group}:${p.ext.artifactId}:${p.version}"
```

Comment:
Should we raise an `IllegalStateException` if `bomGroups` is defined and a `bom` flag is set to a non `bomGroups` project to prevent a mistake?

Reply:
umm, then compatibility will be broken when upgrade `gradle-scripts`. I think that we don't need to raise an `IllegalStateException`.